fastjet/internal/delaunay/triangle.go

```diff
@@ -0,0 +1,190 @@
+	children triangles
+	// A,B,C are the points that make up the triangle
+	A, B, C *Point
+	// inD holds whether a triangle is part of the triangulation
+	inD bool
+}
+
+// NewTriangle returns a triangle formed out of the three given points
+func NewTriangle(a, b, c *Point) *Triangle {
+	// make it counter clockwise
+	if isClockwise(a, b, c) {
+		b, c = c, b
+	}
+	return &Triangle{
+		A: a,
+		B: b,
+		C: c,
+	}
+}
+
+// isClockwise checks whether three points are in clockwise order
+func isClockwise(a, b, c *Point) bool {
+	return (b.Y-a.Y)*(c.X-b.X)-(c.Y-b.Y)*(b.X-a.X) > 0
+}
+
+// inCircumcircle returns whether the point is inside the circumcircle of the triangle
+func (t *Triangle) inCircumcircle(p *Point) bool {
+	x, y := t.centerOfCircumcircle()
+	// use the math/big package to handle the geometric predicates
+	// r is the squared radius of the circumcircle
+	r := big.NewFloat((x-t.A.X)*(x-t.A.X) + (y-t.A.Y)*(y-t.A.Y))
+	// d is the squared distance from the point to the circumcenter
+	d := big.NewFloat((p.X-x)*(p.X-x) + (p.Y-y)*(p.Y-y))
+	// point is in circumcircle if squared distance to center is less than squared radius
+	return d.Cmp(r) < 0
+}
+
+// centerOfCircumcircle returns the center of the triangle's circumcircle.
+func (t *Triangle) centerOfCircumcircle() (x, y float64) {
+	m1 := (t.A.X - t.B.X) / (t.B.Y - t.A.Y)
+	m2 := (t.A.X - t.C.X) / (t.C.Y - t.A.Y)
+	b1 := (t.A.Y+t.B.Y)*0.5 - m1*(t.A.X+t.B.X)*0.5
+	b2 := (t.A.Y+t.C.Y)*0.5 - m2*(t.A.X+t.C.X)*0.5
+	// x and y are the coordinates of the center of the circumcircle
+	x = (b2 - b1) / (m1 - m2)
+	y = m1*x + b1
+	if math.IsNaN(y) {
+		m1 = (t.B.X - t.A.X) / (t.A.Y - t.B.Y)
+		m2 = (t.B.X - t.C.X) / (t.C.Y - t.B.Y)
+		b1 = (t.B.Y+t.A.Y)*0.5 - m1*(t.B.X+t.A.X)*0.5
+		b2 = (t.B.Y+t.C.Y)*0.5 - m2*(t.B.X+t.C.X)*0.5
+		x = (b2 - b1) / (m1 - m2)
+		y = m1*x + b1
+		if math.IsNaN(y) {
+			m1 = (t.C.X - t.A.X) / (t.A.Y - t.C.Y)
+			m2 = (t.C.X - t.B.X) / (t.B.Y - t.C.Y)
+			b1 = (t.C.Y+t.A.Y)*0.5 - m1*(t.C.X+t.A.X)*0.5
+			b2 = (t.C.Y+t.B.Y)*0.5 - m2*(t.C.X+t.B.X)*0.5
+			x = (b2 - b1) / (m1 - m2)
+			y = m1*x + b1
+			if math.IsNaN(y) {
+				panic(fmt.Errorf("delaunay: error caluclating the circumcenter of triangle " + t.String("T")))
+			}
+		}
+	}
+	return x, y
+}
+
+// Equals checks whether two triangles are the same
+func (t *Triangle) Equals(s *Triangle) bool {
+	return t == s ||
+		(t.A.Equals(s.A) || t.A.Equals(s.B) || t.A.Equals(s.C)) &&
+			(t.B.Equals(s.A) || t.B.Equals(s.B) || t.B.Equals(s.C)) &&
+			(t.C.Equals(s.A) || t.C.Equals(s.B) || t.C.Equals(s.C))
+}
+
+func (t *Triangle) String(name string) string {
+	return fmt.Sprintf("{A%s, B%s, C%s}", t.A, t.B, t.C)
+}
+
+type triangles []*Triangle
+
+// appendT appends to a slice of triangles and updates the nearest neighbor
+// it is used when the adjacent triangles of a point change
+func (triangles triangles) appendT(elems ...*Triangle) []*Triangle {
```

Comment:
Please make the receiver simply `t`. `elems` could be renamed `triangles` since the type is already dealt with.

`s/appendT/append/g`; the signature tells people what is being appended.